fake_process is incompatible with anyio

fake_process appears to be incompatible with code that uses anyio with asyncio.

Example test:
```python
def test_anyio(fake_process):
    async def impl():
        await anyio.sleep(1)

    asyncio.get_event_loop().run_until_complete(impl())
```

Error:
```
/usr/lib/python3.8/asyncio/base_events.py:616: in run_until_complete
    return future.result()
tmp/async_devnull_test.py:28: in impl
    await anyio.sleep(1)
/usr/local/lib/python3.8/dist-packages/anyio/_core/_eventloop.py:69: in sleep
    return await get_asynclib().sleep(delay)
/usr/local/lib/python3.8/dist-packages/anyio/_core/_eventloop.py:140: in get_asynclib
    return import_module(modulename)
/usr/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1014: in _gcd_import
    ???
<frozen importlib._bootstrap>:991: in _find_and_load
    ???
<frozen importlib._bootstrap>:975: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:671: in _load_unlocked
    ???
/usr/local/lib/python3.8/dist-packages/_pytest/assertion/rewrite.py:170: in exec_module
    exec(co, module.__dict__)
/usr/local/lib/python3.8/dist-packages/anyio/_backends/_asyncio.py:897: in <module>
    class Process(abc.Process):
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

    @dataclass(eq=False)
    class Process(abc.Process):
>       _process: asyncio.subprocess.Process
E       AttributeError: module 'pytest_subprocess.asyncio_subprocess' has no attribute 'Process'
```

Notably this only occurs if nothing else has caused `anyio._backends._asyncio` to be imported yet and can be worked around with a fixture like this (or by otherwise running the equivalent code before the fake_process fixture runs):
```python
@pytest.fixture
def init_anyio():
    async def impl():
        anyio.get_cancelled_exc_class()

    asyncio.get_event_loop().run_until_complete(impl())
```

FWIW the example test case is not representative of the actual test we ran into it with, but just something that reproduces the problem. In the real test we're testing a FastAPI-based server to see if it invokes processes when it should, and FastAPI is based on anyio.

--- a/changelog.d/bug.baf01777.entry.yaml
+++ b/changelog.d/bug.baf01777.entry.yaml
@@ -1,5 +1,6 @@
-message: Add missing consts to `asyncio.subprocess`.
+message: Add missing items to `asyncio.subprocess`.
 pr_ids:
 - '63'
+- '67'
 timestamp: 1639330523
 type: bug

--- a/pytest_subprocess/asyncio_subprocess.py
+++ b/pytest_subprocess/asyncio_subprocess.py
@@ -1,5 +1,7 @@
 from asyncio.subprocess import DEVNULL
 from asyncio.subprocess import PIPE
+from asyncio.subprocess import Process
 from asyncio.subprocess import STDOUT
+from asyncio.subprocess import SubprocessStreamProtocol
 
-_ = (DEVNULL, PIPE, STDOUT)
+_ = (DEVNULL, PIPE, STDOUT, Process, SubprocessStreamProtocol)

--- a/setup.py
+++ b/setup.py
@@ -40,6 +40,7 @@ setup(
             "Pygments>=2.0",
             "pytest-rerunfailures",
             "pytest-asyncio>=0.15.1",
+            "anyio",
         ],
         "dev": ["nox", "changelogd"],
         "docs": [

--- a/tests/test_asyncio.py
+++ b/tests/test_asyncio.py
@@ -3,6 +3,7 @@ import os
 import sys
 import time
 
+import anyio
 import pytest
 
 
@@ -184,6 +185,11 @@ async def test_devnull_stdout(fake_process):
     fake_process.register_subprocess("cat")
 
     await impl()
+
+
+@pytest.mark.asyncio
+async def test_anyio(fake_process):
+    await anyio.sleep(0.01)
 
 
 @pytest.fixture(autouse=True)

